src/sagas/reown.js

```diff
@@ -0,0 +1,996 @@
+            data: request.data,
+            dapp: requestMetadata,
+          }
+        });
+      } break;
+
+      case TriggerTypes.CreateTokenConfirmationPrompt: {
+        const createTokenResponseTemplate = (accepted) => (data) => {
+          dispatch(hideGlobalModal());
+          resolve({
+            type: TriggerResponseTypes.CreateTokenConfirmationResponse,
+            data: {
+              accepted,
+              token: data?.payload,
+            }
+          });
+        };
+
+        dispatch({
+          type: types.SHOW_CREATE_TOKEN_REQUEST_MODAL,
+          payload: {
+            accept: createTokenResponseTemplate(true),
+            deny: createTokenResponseTemplate(false),
+            data: request.data,
+            dapp: requestMetadata,
+          }
+        });
+      } break;
+
+      case TriggerTypes.SignMessageWithAddressConfirmationPrompt: {
+        const signMessageResponseTemplate = (accepted) => () => {
+          dispatch(hideGlobalModal());
+          resolve({
+            type: TriggerResponseTypes.SignMessageWithAddressConfirmationResponse,
+            data: accepted,
+          });
+        };
+
+        dispatch({
+          type: types.SHOW_SIGN_MESSAGE_REQUEST_MODAL,
+          payload: {
+            accept: signMessageResponseTemplate(true),
+            deny: signMessageResponseTemplate(false),
+            data: request.data,
+            dapp: requestMetadata,
+          }
+        });
+      } break;
+
+      case TriggerTypes.SendNanoContractTxLoadingTrigger:
+        dispatch(setNewNanoContractStatusLoading());
+        dispatch(showGlobalModal(MODAL_TYPES.NANO_CONTRACT_FEEDBACK, { isLoading: true }));
+        resolve();
+        break;
+
+      case TriggerTypes.CreateTokenLoadingTrigger:
+        log.debug('CreateTokenLoadingTrigger: Starting token creation process');
+        dispatch(setCreateTokenStatusLoading());
+        dispatch(showGlobalModal(MODAL_TYPES.TOKEN_CREATION_FEEDBACK, { isLoading: true }));
+        resolve();
+        break;
+
+      case TriggerTypes.CreateTokenLoadingFinishedTrigger:
+        log.debug('CreateTokenLoadingFinishedTrigger: Token creation process completed');
+        dispatch(setCreateTokenStatusReady());
+        dispatch(hideGlobalModal());
+        resolve();
+        break;
+
+      case TriggerTypes.SendNanoContractTxLoadingFinishedTrigger:
+        dispatch(setNewNanoContractStatusReady());
+        resolve();
+        break;
+
+      case TriggerTypes.PinConfirmationPrompt: {
+        const pinPromise = new Promise((pinResolve, pinReject) => {
+          dispatch(showGlobalModal(MODAL_TYPES.PIN_PAD, {
+            onComplete: (pinCode) => {
+              dispatch(hideGlobalModal());
+              pinResolve(pinCode);
+            },
+            onCancel: () => {
+              dispatch(hideGlobalModal());
+              dispatch(setCreateTokenStatusReady()); // Reset loading state if PIN entry is cancelled
```

Comment:
What if it's a nano contract creation or another method? No need to set the status to ready?

Reply:
Sorry, this code doesn't make sense

Refatored it to reset state on modals' unmount